internal/router/router.go

```diff
@@ -151,17 +167,23 @@ type Router struct {
 // The exists function may be called to test whether an entity
 // exists when an API endpoint needs to know that
 // but has no appropriate handler to call.
+//
+// The authorizeAdmin function will be called to authorize admin
+// requests. The Cause of the authorizeAdmin error will be left
+// unchanged, as for the handlers.
 func New(
 	handlers *Handlers,
 	resolveURL func(id *charm.Reference) error,
 	authorize func(id *charm.Reference, req *http.Request) error,
 	exists func(id *charm.Reference, req *http.Request) (bool, error),
+	authorizeWithPerms func(req *http.Request, read, write []string) error,
```

Comment:
I wonder if rather than having two functions with similar but overlapping behaviour, maybe we could have two functions that don't overlap so much, say:

    permsForEntity func(id *charm.Reference) ([]string, error)
    authorize func(req *http.Request, read, write []string) error

The first one would be called any time we'd previously call authorize.
The second one would be called for any authorization request.

This opens the door for permissions on other kinds of entity with a
reasonably uniform mechanism.